### PR TITLE
Fix some linker annotations in S.CM.TypeConverter

### PR DESCRIPTION
**Pull request by pgovind**
For https://github.com/dotnet/runtime/pull/48527#discussion_r605216290

cc @jkotas

### CODE REVIEW

File: src/libraries/System.ComponentModel.TypeConverter/src/ILLink/ILLink.Suppressions.LibraryBuild.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    </attribute>
+    <attribute fullname="System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute">
+      <argument>ILLink</argument>
+      <argument>IL2026</argument>
+      <property name="Scope">member</property>
+      <property name="Target">M:System.ComponentModel.Design.DesigntimeLicenseContextSerializer.DeserializeUsingBinaryFormatter(System.ComponentModel.Design.DesigntimeLicenseContextSerializer.StreamWrapper,System.String,System.ComponentModel.Design.RuntimeLicenseContext)</property>
+      <property name="Justification">This warning is left in the product so developers get an ILLink warning when trimming an app with System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization=true.</property>
+    </attribute>
+  </assembly>
+</linker
```
Comment (safern) [inline src/libraries/System.ComponentModel.TypeConverter/src/ILLink/ILLink.Suppressions.LibraryBuild.xml:19]: Missing close token:

```suggestion
</linker>
```

Comment (pgovind) [reply src/libraries/System.ComponentModel.TypeConverter/src/ILLink/ILLink.Suppressions.LibraryBuild.xml:19]: Indeed. Just fixed that locally :)

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs [hunk 2]
```diff
@@ -50,8 +50,6 @@ public static void Serialize(Stream o, string cryptoKey, DesigntimeLicenseContex
             }
         }
 
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
```
Comment (pgovind) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs:53]: @eerhardt : You also made some infra changes so mscorelib.dll would exist after trimming in https://github.com/dotnet/runtime/pull/48527. I don't really understand the changes, so do we keep them or do you want me to remove those changes?

**Review by jkotas — APPROVED** [review]
Thank you!

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs (hunk 2 above)
Comment (eerhardt) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs:53]: Remove them please. We can add them back if/when we need them.

**Review by eerhardt — APPROVED** [review]
Thanks @pgovind

**Pull request merged**